platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/Retry.java

```diff
@@ -0,0 +1,88 @@
+     **/
+    private Retry() {}
+
+    /**
+     * Evaluates a given {@code value} using the provided {@code checkFn} method until the return value is {@code true}
+     * or the {@code maxAttempts} are exhausted. A delay of {@code delayMs} will be applied between each call to the
+     * {@code checkFn} method. Setting the {@code delayMs} parameter to zero (0) will disable the delay mechanism and
+     * the {@code checkFn} method will be called as fast as possible.
+     *
+     * @param value       the user value to be passed to the {@code checkFn} method.
+     * @param maxAttempts the maximum number of retry attempts.
+     * @param delayMs     the delay between retry attempts. must greater than or equal to zero (0). if a zero value is
+     *                    specified then no delay will be applied.
+     * @param <T>         the type of the value argument.
+     * @return true if the {@code checkFn} method returns true before all attempts have been exhausted; false
+     * if the retries are exhausted.
+     * @throws NullPointerException     if the {@code checkFn} or the {@code value} arguments are a {@code null} values.
+     * @throws IllegalArgumentException if the {@code maxAttempts} argument is less than or equal to zero (0) or the
+     *                                  {@code delayMs} argument is less than zero (0).
+     * @throws ExecutionException       if an unhandled {@link RuntimeException} was thrown by the {@code checkFn}
+     *                                  method call.
+     */
+    public static <T> boolean check(
+            @NonNull Function<T, Boolean> checkFn, @NonNull final T value, final int maxAttempts, final int delayMs)
+            throws InterruptedException, ExecutionException {
+        Objects.requireNonNull(checkFn, "checkFn must not be null");
+        Objects.requireNonNull(value, "value must not be null");
+
+        if (maxAttempts <= 0) {
+            throw new IllegalArgumentException("The maximum number of attempts must be greater than zero (0)");
+        }
+
+        if (delayMs < 0) {
+            throw new IllegalArgumentException("The delay must be greater than or equal to zero (0)");
+        }
+
+        try {
+            for (int i = 0; i < maxAttempts; i++) {
+                if (checkFn.apply(value)) {
+                    return true;
+                }
+
+                if (delayMs > 0) {
+                    TimeUnit.MILLISECONDS.sleep(delayMs);
+                }
+            }
+        } catch (final Throwable e) {
+            // If an exception other than UnknownHostException or InterruptedException is thrown, then we should
+            // immediately skip any further retry attempts.
+            throw new ExecutionException("Unhandled exception thrown during a retryable operation attempt", e);
+        }
+
+        // If we failed to resolve the name to an IP address within the maximum retry threshold, then we should fall
+        // through to here to fail.
```

Comment:
The comment in the body (line 79-80, 84-85) needs to be polished a bit without referring to resolving an IP or UnknownHostException